uat/custom-components/client-mosquitto-c/src/MqttConnection.cpp

```diff
@@ -0,0 +1,591 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cstdio>
+#include <string>
+#include <sstream>
+#include <iterator>
+#include <memory>
+#include <future>
+
+
+#include <mosquitto.h>                          /* mosquitto_new() mosquitto_destroy() mosquitto_connect_bind_v5() ... */
+#include <mqtt_protocol.h>                      /* MQTT_RC_DISCONNECT_WITH_WILL_MSG */
+
+#include "mqtt_client_control.grpc.pb.h"
+
+#include "MqttConnection.h"                     /* self header */
+#include "MqttException.h"
+#include "logger.h"                             /* logd() */
+
+#define DEFAULT_DISCONNECT_REASON       MQTT_RC_NORMAL_DISCONNECTION
+#define DEFAULT_DISCONNECT_TIMEOUT      10
+
+#define CONNECT_REQUST_ID               (1024*64 + 1)
```

Comment:
REQUEST

Reply:
fixed in 8106e75fbe54a32c4b22de1157aadbf12ffe3a5b